src/api/app.py

```diff
@@ -61,75 +67,82 @@ async def close_async_db_connection(app: Litestar) -> None:
-        """
-        if session_id not in self.file_handles:
-            write_path = os.path.join(self.DEMOS_PATH, f"{session_id}.dem")
-            self.file_handles[session_id] = open(write_path, "wb")
+    async with engine.connect() as conn:
+        result = await conn.execute(
+            sa.text(sql),
+            params,
+        )
 
-        return self.file_handles[session_id]
-
-    def handle_demo_data(self, data: dict[str, str | bytes | int]) -> None:
-        """Handle incoming data from a client upload.
+        data = result.all()
+        is_active = bool(data)
 
-        Args:
-            data: dict of {session_id: ..., data: bytes or self.SENTINEL}
-        """
-        session_id = data["session_id"]
+        return is_active
 
-        file_handle = self.make_or_get_file_handle(session_id)
 
-        _data = base64.b64decode(data["data"])
+async def valid_key_guard(connection: ASGIConnection, _: BaseRouteHandler) -> None:
+    """A Guard clause to validate the user's API key."""
+    api_key = connection.query_params["api_key"]
 
-        if _data == self.SENTINEL:
-            file_handle.close()
-        else:
-            file_handle.write(_data)
-            file_handle.flush()
+    async_engine = connection.app.state.async_engine
+    exists = await _check_key_exists(async_engine, api_key)
+    if not exists:
+        raise NotAuthorizedException()
 
-        def close(self, session_id: int) -> None:
-            self.file_handles[session_id].close()
 
+async def user_in_session_guard(connection: ASGIConnection, _: BaseRouteHandler) -> None:
+    """Assert that the user is not currently in a session."""
+    async_engine = connection.app.state.async_engine
 
-demo_manager = DemoSessionManager()
+    api_key = connection.query_params["api_key"]
+    is_active = await _check_is_active(async_engine, api_key)
 
+    if is_active:
+        raise NotAuthorizedException(
+            detail="User is already in a session, either remember your session token or close it out at `/close_session`!"  # noqa
+        )
 
-def generate_uuid4_int() -> int:
-    """Seems useless, but makes testing easier."""
-    return uuid4().int
 
+async def user_not_in_session_guard(connection: ASGIConnection, _: BaseRouteHandler) -> None:
+    """Assert that the user is not currently in a session."""
+    async_engine = connection.app.state.async_engine
 
-@get("/session_id", guards=[valid_key_guard], sync_to_thread=False)
-def session_id(api_key: str) -> dict[str, int]:
+    api_key = connection.query_params["api_key"]
+    session_id = connection.query_params["session_id"]
+    is_active = await _check_is_active(async_engine, api_key, session_id)
+    if not is_active:
+        raise NotAuthorizedException(
+            detail="User is not in a session, create one at `/session_id`!"  # noqa
+        )
+
+
+@get("/session_id", guards=[valid_key_guard, user_in_session_guard], sync_to_thread=False)
+def session_id(
```

Comment:
Make sure the string is formatted such that it is not shortened like firefox does.

Reply:
changed typing to be a string in the annotations which will reflect on the endpoint, good catch here